src/solvers/flattening/boolbv.cpp

```diff
@@ -505,53 +512,56 @@ literalt boolbvt::convert_rest(const exprt &expr)
     return convert_overflow(expr);
   else if(expr.id()==ID_isnan)
   {
-    DATA_INVARIANT(expr.operands().size() == 1, "isnan expects one operand");
-    const bvt &bv=convert_bv(operands[0]);
+    const auto &op = to_unary_expr(expr).op();
+    const bvt &bv = convert_bv(op);
```

Comment:
It seems the original code had `operands` declared somewhere. In the code review it's hard to see where that is, but can you please make sure its declaration is removed as it shouldn't be of any use anymore?